Source/WebCore/platform/graphics/ComplexTextController.cpp

```diff
@@ -314,7 +314,21 @@ void ComplexTextController::collectComplexTextRuns()
         return characters;
     }();
 
+    auto isJapanese = [](const String& str) {
+        StringView textView = str;
+        for (size_t i = 0; i < textView.length(); ++i) {
+            auto ch = textView[i];
+            if ((ch >= 0x4E00 && ch <= 0x9FFF) || (ch >= 0x30A0 && ch <= 0x30FF) || (ch >= 0x4E00 && ch <= 0x9FFF))
+                return true;
+        }
+        return false;
+    };
+
     auto fontVariantCaps = m_font.fontDescription().variantCaps();
+    bool isSmall = fontVariantCaps == FontVariantCaps::Small || fontVariantCaps == FontVariantCaps::AllSmall;
+    if (isJapanese(m_run.textAsString()) && isSmall)
```

Comment:
please use complete meaningful names: "isSmallCapsVariant" or something like that. Same for `isJapanese`